model/validation.go

```diff
@@ -506,17 +507,10 @@ func ValidateCIDRBlock(cidr string) error {
 
 // ValidateIPv4Address validates that a string in "CIDR notation" or "Dot-decimal notation"
 func ValidateIPv4Address(addr string) error {
-	octets := strings.Split(addr, ".")
-	if len(octets) != 4 {
+	err := net.ParseIP(addr)
```

Comment:
s/`err`/`ip`/. The return value is of type `net.IP` and not error.

Also, we should check if `ip` is IPv4 if existing semantics are to be preserved. Otherwise we'll allow IPv6.

Reply:
Is it necessary to also validate the subnet mask? I noted that we have a separate function for it and the test case here https://github.com/cmluciano/pilot/blob/c9ce6417fbac824ad505e6f07f48cab88b6e9690/model/validation_test.go#L481 does not make sense. 

We specify a parameter for SourceSubnet: []string{"1.2.3.4"} but we do not mark the subnet mask explicitly with proper CIDR notation. Is this test case a subtly bug or do we not care about passing the explicit mask and only care if the IP address is IPv4?

Reply:
The subnet is optional (see [here](https://github.com/istio/api/blob/93848bb1c495ff6ebc1a938e2e0a403d52d345c1/proxy/v1/config/route_rule.proto#L269)). `ValidateIPv4Subnet` validates both cases.

Reply:
Ok, I can migrate the ValidateIPv4Subnet to use Go net libraries for doing more of this validation in this PR as well. WDYT?

Reply:
sgtm